src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/DwpUploadResponseHandler.java

```diff
@@ -47,13 +48,42 @@ public void handle(CallbackType callbackType, Callback<SscsCaseData> callback) {
             throw new IllegalStateException("Cannot handle callback");
         }
 
-        boolean dwpFurtherInfo =
+        BenefitType benefitType = callback.getCaseDetails().getCaseData().getAppeal().getBenefitType();
+        log.info("BenefitType" + benefitType);
+
+        if (benefitType.getCode().equals("UC")) {
+            handleUc(callbackType, callback);
+        } else {
+            handleNonUc(callbackType, callback);
+        }
+    }
+
+    private void handleNonUc(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        boolean notDwpFurtherInfo =
             StringUtils.equalsIgnoreCase(callback.getCaseDetails().getCaseData().getDwpFurtherInfo(), "no");
-        boolean disputedDecision =
-            StringUtils.equalsIgnoreCase(callback.getCaseDetails().getCaseData().getElementsDisputedIsDecisionDisputedByOthers(), "no");
 
-        if (dwpFurtherInfo
-            && disputedDecision) {
+        if (notDwpFurtherInfo) {
+            log.info("updating to ready to list");
+
+            SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+
+            caseData.setDwpState(DwpState.RESPONSE_SUBMITTED_DWP.getId());
```

Comment:
This is repeated code along with the updateCase, would be good to put this in a method